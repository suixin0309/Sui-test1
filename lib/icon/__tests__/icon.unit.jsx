import * as React from "react"
import * as renderer from "react-test-renderer"
import {mount} from 'enzyme'
import Icon from "../icon"

describe('icon用例', () => {
    it('render successfully', () => {
        const json = renderer.create(<Icon name='alipay'/>).toJSON()
        expect(json).toMatchSnapshot()
    });
    it('onClick', function () {
        const fn=jest.fn()
        const fn2=jest.fn()   //
        const compontent =mount(<Icon name='alipay' onClick={fn}/>)
        compontent.find('svg').simulate('click')
        expect(fn).toBeCalled()
        // expect(fn2).toBeCalled()   错误样例


    });
});