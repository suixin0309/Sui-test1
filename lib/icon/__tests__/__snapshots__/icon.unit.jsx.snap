// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon用例 是div 1`] = `
<svg
  className="sui-icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
